src/main/java/io/quarkus/bot/PingWhenNeedsTriageRemoved.java

```diff
@@ -51,29 +55,22 @@ void pingWhenNeedsTriageRemoved(@Issue.Unlabeled GHEventPayload.Issue issuePaylo
             }
         }
 
-        if (removedLabel.getName().equals(Labels.TRIAGE_NEEDS_TRIAGE)) {
-            if (!mentions.isEmpty()) {
-                if (!quarkusBotConfig.isDryRun()) {
-                    issue.comment("/cc @" + String.join(", @", mentions));
-                } else {
-                    LOG.info("Issue #" + issue.getNumber() + " - Ping: " + mentions);
-                }
-            }
+        if (mentions.isEmpty()) {
+            return;
+        }
+
+        if (!quarkusBotConfig.isDryRun()) {
+            issue.comment("/cc @" + String.join(", @", mentions));
+        } else {
+            LOG.info("Issue #" + issue.getNumber() + " - Ping: " + mentions);
         }
     }
 
     private static boolean matchRule(GHIssue issue, QuarkusBotConfigFile.TriageRule rule) {
-
-        try {
-            if (Strings.isNotBlank(rule.labels.toString())) {
-                if (Patterns.find(rule.labels.toString(), issue.getLabels().toString())) {
-                    return true;
-                }
-            }
-        } catch (Exception e) {
-            LOG.error("Error evaluating labels: " + rule.labels, e);
+        if (rule.labels.isEmpty() || rule.notify.isEmpty()) {
+            return false;
         }
 
-        return false;
+        return issue.getLabels().stream().anyMatch(l -> rule.labels.contains(l.getName()));
```

Comment:
Here I use a bit of lambda magic and you could do it with simple loops but the point is that I compare the name of each label, I don't join them as a big string.

Note that your approach had 2 issues: it might not work at all depending on the strings as you're losing the structure of the data and it's slower as pattern matching is slow. Here we don't want to do pattern matching, we want to match exactly.

Reply:
Thanks for the explanation, Guillaume!

Just a few hours ago I was trying to understand all these on my playground project. 
It's all very clear now after looking at your commit.


![image](https://user-images.githubusercontent.com/46818757/114276538-ab75a680-9a44-11eb-8bda-fb59a561ebcb.png)